contracts/staking/StakingPoolMediator.sol

```diff
@@ -0,0 +1,201 @@
+     */
+    function initialize(StakingPoolFactory factory, address treasury)
+        external
+        initializer
+    {
+        require(
+            AddressUpgradeable.isContract(address(factory)),
+            "SPM: creator not a contract"
+        );
+
+        __StakingPoolCurator_init();
+        __DaoConfiguration_init();
+        __UUPSUpgradeable_init();
+        __TokenSweep_init(treasury);
+
+        _creator = factory;
+    }
+
+    function createDao(address erc20CapableTreasury)
+        external
+        atLeastDaoCreatorRole
+        returns (uint256)
+    {
+        uint256 id = _daoConfiguration(erc20CapableTreasury);
+        _grantDaoCreatorAdminRoleInTheirDao(id);
+
+        emit CreateDao(id, erc20CapableTreasury, _msgSender());
+
+        return id;
+    }
+
+    function createStakingPool(
+        StakingPoolLib.Config calldata config,
+        bool launchPaused,
+        uint32 rewardsAvailableTimestamp
+    )
+        external
+        whenNotPaused
+        atLeastDaoMeepleRole(config.daoId)
+        returns (address)
+    {
+        require(_isValidDaoId(config.daoId), "SPM: invalid DAO Id");
+        require(
+            isAllowedDaoCollateral(config.daoId, address(config.stakeToken)),
+            "SPM: collateral not whitelisted"
+        );
+
+        // Reentrancy warning from an emitted event, which needs the Bond, created by an external call above.
+        //slither-disable-next-line reentrancy-events
+        address stakingPool = _creator.createStakingPool(
+            config,
+            launchPaused,
+            rewardsAvailableTimestamp
+        );
+
+        _addStakingPool(config.daoId, stakingPool);
+
+        return stakingPool;
+    }
+
+    /**
+     * @notice Updates the Bond creator reference.
+     *
+     * @param factory Contract address for the new BondCreator to use from now onwards when creating managed bonds.
+     */
+    function setStakingPoolFactory(address factory)
+        external
+        whenNotPaused
+        atLeastSysAdminRole
+    {
+        require(
+            AddressUpgradeable.isContract(factory),
+            "SPM: creator not a contract"
+        );
+        address previousCreator = address(_creator);
+        require(factory != previousCreator, "BM: matches existing");
+
+        emit StakingPoolFactoryUpdate(
+            address(_creator),
+            address(factory),
+            _msgSender()
+        );
+        _creator = StakingPoolFactory(factory);
+    }
+
+    /**
+     * @notice Permits updating the default DAO treasury address.
+     *
+     * @dev Only applies for bonds created after the update, previously created bond treasury addresses remain unchanged.
+     */
+    function setDaoTreasury(uint256 daoId, address replacement)
+        external
+        whenNotPaused
+        atLeastDaoAdminRole(daoId)
+    {
+        _setDaoTreasury(daoId, replacement);
+    }
+
+    function updateTokenSweepBeneficiary(address newBeneficiary)
```

Comment:
@CjHare some of this functionality is repeated between `BondMediator` and `StakingPoolMediator` - would it be worth creating a base `Mediator` contract or is this too complex for access control

Reply:
would have the advantage of having commonality in testing -- `stakingpool-mediator.test.ts` does not yet exist but would look similar to `bond-mediator.test.ts`

Reply:
It depends.

How much commonality is there really?

Does this refactoring need to be done now, or could it be done later?

How much testing effort (in terms of days) would be saved the `Mediator` contract?